SMIT CT Lung GTV segmentation model

locastre opened the pull request:
Submitting the SMIT CT Lung GTV segmentation model for mHub

locastre committed:
Create Dockerfile for MSK SMIT Lung GTV

locastre committed:
1st draft Dockerfile

locastre committed:
Create README.md

locastre committed:
draft SMITrunner.py

locastre committed:
add src files

locastre committed:
add utils

locastre committed:
Add files via upload

locastre committed:
Add files via upload

locastre committed:
Update Dockerfile

locastre committed:
mkdir for conda-pack

locastre committed:
add default.yml

locastre committed:
add config.json

locastre committed:
Rename config.json to meta.json

locastre committed:
Update meta.json

locastre committed:
Update meta.json

locastre committed:
Update meta.json

locastre committed:
Update meta.json

locastre committed:
Update meta.json

locastre committed:
Update Dockerfile

locastre committed:
Update SMITrunner.py

LennyN95 commented on models/msk_smit_lung_gtv/dockerfiles/Dockerfile:
```diff
@@ -0,0 +1,24 @@
+
+WORKDIR ${WORK_DIR}/msk_smit_lung_gtv/src
+ENV WEIGHTS_URL=https://mskcc.box.com/shared/static/sf7jic4m2dk67413cipbbq6hddvhpj61.gz
+ENV CONDA_URL=https://mskcc.box.com/shared/static/d580gfjzzmt26v8klwp8pivb6wafomag.gz
```
Is this using a conda environment for dependencies?

We should avoid conda in open source projects due to it's licensing scheme. 

I suggest we install all dependencies with uv (which comes with our base image), e.g., you could provide a pyproject.toml to make src installable or just provide a list with dependencies (like a requirements.txt) and install with `uv pip install`.

locastre replied:
Thank you @LennyN95 . Our updated repo includes a requirements.txt to set up Python 3.9 in a virtualenv

LennyN95 commented on models/msk_smit_lung_gtv/meta.json:
```diff
@@ -0,0 +1,102 @@
+{
+    "id": "", 
+    "name": "msk_smit_lung_gtv", 
+    "title": "CT Lung GTV SMIT Segmentation",
```
What does SMIT stand for :)?

locastre replied:
SMIT = "Self-distilled Masked Image Transformer", has been added to the extended name in the details section (line 46)

LennyN95 commented on models/msk_smit_lung_gtv/meta.json:
```diff
@@ -0,0 +1,102 @@
+    "name": "msk_smit_lung_gtv", 
+    "title": "CT Lung GTV SMIT Segmentation",
+    "summary": {
+    "description": "GTV segmentation from CT scan",
```
Could be slightly longer, this is what the user sees on the website.

locastre replied:
Augmented the title and description:
```
    "title": "Self-supervised 3D segmentation using self-distilled masked image transformer for Lung GTV Segmentation",
    "summary": {
    "description": "A Lung GTV segmentation model, fine-tuned from a foundation model pretrained with 10K CT scans",
```

LennyN95 commented on models/msk_smit_lung_gtv/meta.json:
```diff
@@ -0,0 +1,102 @@
+            "description": "Segmentation of the lung GTV from NIfTI CT images.",
+            "type": "Segmentation",                       
+            "classes": [  
+                "GTV"
```
See comment in SMITrunner.py.

locastre replied:
renamed segmentation class to LUNG+NEOPLASM_MALIGNANT_PRIMARY

LennyN95 commented on models/msk_smit_lung_gtv/meta.json:
```diff
@@ -0,0 +1,102 @@
+        "version": "1.0.0", 
+        "devteam": "",  
+        "authors": ["Jue Jiang, Harini Veeraraghavan"],      
+        "type": "it is a 3D Swin transformer based segmentation net",   
```
Just `Segmentation`?

locastre replied:
Simplified the `type` entry to match `"Segmentation"`

LennyN95 commented on models/msk_smit_lung_gtv/meta.json:
```diff
@@ -0,0 +1,102 @@
+        "authors": ["Jue Jiang, Harini Veeraraghavan"],      
+        "type": "it is a 3D Swin transformer based segmentation net",   
+        "date": {
+            "code": "11.03.2025",   
+            "weights": "11.03.2025",
+            "pub": "15.July.2024"    
```
Must be `dd.mm.yyyy` consistently.

locastre replied:
corrected the July date formatting

LennyN95 commented on models/msk_smit_lung_gtv/meta.json:
```diff
@@ -0,0 +1,102 @@
+		},
+		"training": {
+			"title": "Training data",
+			"text": "Training data was from 377 data in the TCIA NSCLC-Radiomics data, references: Aerts, H. J. W. L., Wee, L., Rios Velazquez, E., Leijenaar, R. T. H., Parmar, C., Grossmann, P., Carvalho, S., Bussink, J., Monshouwer, R., Haibe-Kains, B., Rietveld, D., Hoebers, F., Rietbergen, M. M., Leemans, C. R., Dekker, A., Quackenbush, J., Gillies, R. J., Lambin, P. (2014). Data From NSCLC-Radiomics  ...
```
Please add references using the `references: []` key, the format is described [here](https://github.com/MHubAI/documentation/blob/main/documentation/mhub_models/model_json.md#info).

LennyN95 commented on models/msk_smit_lung_gtv/meta.json:
```diff
@@ -0,0 +1,102 @@
+			"text": "Training data was from 377 data in the TCIA NSCLC-Radiomics data, references: Aerts, H. J. W. L., Wee, L., Rios Velazquez, E., Leijenaar, R. T. H., Parmar, C., Grossmann, P., Carvalho, S., Bussink, J., Monshouwer, R., Haibe-Kains, B., Rietveld, D., Hoebers, F., Rietbergen, M. M., Leemans, C. R., Dekker, A., Quackenbush, J., Gillies, R. J., Lambin, P. (2014). Data From NSCLC-Radiomics  ...
+	
+		},
+    	"analyses": {
+			"title": "Quantitative Analyses",
+			"text": "DSC was used to compute the accuracy of the model"
+		},
```
Extend.

locastre replied:
We've fleshed out this section with additional details and references
```
"analyses": {
			"title": "Evaluation",
			"text": "Evaluation was determined with DICE score, See the paper (Methods, Section 4.2, section on Experiments and evaluation metrics, and Results 5.1, Table 2 for additional details.",
			"references": [
				 {
				  "label": "Self-supervised pretraining in the wild imparts image acquisition robustness to medical image transformers: an application to lung cancer segmentation",
				  "uri": "https://proceedings.mlr.press/v250/jiang24b.html"
				}
			  ],
```

LennyN95 replied:
Great. Ideally, we could include the table here (see [this section](https://github.com/MHubAI/documentation/blob/main/documentation/mhub_models/model_json.md#info) in our documentation):

```text
"/info/*": {            // description|use|analyses|evaluation|training|ethics|limitations
    "text": "",         // str: the text of the section
    "references": [{    // list[str]: optional list of references (e.g., links)
        "label": "",    // str: the label of the reference
        "uri": ""       // str: the url of the reference
    }],
    "tables": [{        // list[obj]: optional list of tables
        "label": "",    // str: the label of the table
        "entries": {}   // obj[str->str]: table rows, the key will appear in the first, the value in the second column
    }]
}
```

LennyN95 commented on models/msk_smit_lung_gtv/meta.json:
```diff
@@ -0,0 +1,102 @@
+    	"evaluation": {
+			"title": "Evaluation data",
+			"text": "To assess the model's segmentation performance in the NSCLC Radiogenomics dataset, we considered that the original input data is a full 3D volume. The model segmented not only the labeled tumor but also tumors that were not manually annotated. Therefore, we evaluated the model based on the manually labeled tumors. After applying the segmentation model, we extracted a 128*128*128 cubic ...
+			"references": [],
+			"tables": ["validation_data_id and DSC value, Validation data is 139 data in the NSCLC Radiogenomics data:https://www.cancerimagingarchive.net/collection/nsclc-radiogenomics/, AMC-001:0.023977216,AMC-005:0.84385232,AMC-006:0.844950109,AMC-011:0.885911774,AMC-013:0.786724403,AMC-014:0.628335342,AMC-016:0.708633094,AMC-019:0.791600435,AMC-020:0.882119609,AMC-021:0.834135707,AMC-022:0.688767807,A ...
```
The format for tables is described [here](https://github.com/MHubAI/documentation/blob/main/documentation/mhub_models/model_json.md#info).

LennyN95 commented on models/msk_smit_lung_gtv/utils/SMITrunner.py:
```diff
@@ -0,0 +1,52 @@
+
+    @IO.Instance()
+    @IO.Input('scan', 'nifti:mod=ct',  the='input ct scan')
+    @IO.Output('gtv_mask', 'gtv_mask.nii.gz', 'nifti:mod=seg:model=SMIT:roi=GTV',data='scan', the='predicted lung gtv')
```
The segmentation ID `GTV` does not exist in [SegDB](https://github.com/MHubAI/SegDB). You can find an overview of available segmentations [here](https://github.com/MHubAI/SegDB/blob/main/segdb/data/segmentations.csv). 

Note, that a tumor is usually described in the context of a parent structure, e.g., segmentation ID `LUNG`, `LEFT_LUNG`, etc. A potential fit could be `LUNG+NEOPLASM_MALIGNANT_PRIMARY`.

LennyN95 commented on models/msk_smit_lung_gtv/utils/SMITrunner.py:
```diff
@@ -0,0 +1,52 @@
+    @IO.Instance()
+    @IO.Input('scan', 'nifti:mod=ct',  the='input ct scan')
+    @IO.Output('gtv_mask', 'gtv_mask.nii.gz', 'nifti:mod=seg:model=SMIT:roi=GTV',data='scan', the='predicted lung gtv')
+    def task(self, instance: Instance, scan: InstanceData, gtv_mask: InstanceData) -> None:
```
The input ct image is available under `scan.abspath`, the segmentation mask generated by the model should be written or copied into `gtv_mask.abspath`.

LennyN95 commented on models/msk_smit_lung_gtv/utils/SMITrunner.py:
```diff
@@ -0,0 +1,52 @@
+    @IO.Output('gtv_mask', 'gtv_mask.nii.gz', 'nifti:mod=seg:model=SMIT:roi=GTV',data='scan', the='predicted lung gtv')
+    def task(self, instance: Instance, scan: InstanceData, gtv_mask: InstanceData) -> None:
+        
+        workDir = os.path.join(os.environ['WORK_DIR'],'models','msk_smit_lung_gtv','src')   # Needs to be defined in docker file as ENV WORK_DIR=path_to_dir e.g. /app/models/SMIT/workDir
```
Could we use `pathlib` for all path concatenations to increase readability?

locastre replied:
Done

LennyN95 commented on models/msk_smit_lung_gtv/utils/SMITrunner.py:
```diff
@@ -0,0 +1,52 @@
+        #condaEnvDir = os.path.join(wrapperInstallDir,'conda-pack')
+        #condaEnvActivateScript = os.path.join(condaEnvDir, 'bin', 'activate')
+        wrapperPath = os.path.join(workDir,'bash_run_SMIT_Segmentation.sh')
+        load_weight_name = os.path.join(workDir,'trained_weights','model.pt')
```
Are there multiple weights trained for this model architecture? Otherwise, I'd suggest we make this parameter fixed, to increase the readability.

locastre replied:
Hard-coded path to weights in SMITRunner

LennyN95 commented on models/msk_smit_lung_gtv/meta.json:
```diff
@@ -0,0 +1,102 @@
+        }
+    ],
+    "model": {
+        "architecture": "Swin Transformer based segmentation, self-supervised pretrained with 10k CT data", 
```
Just `Swin3D`?

locastre replied:
Simplified architecture description to `Swin3D Transformer`

LennyN95 reviewed (commented):
Thank you for your contribution 🚀 

- Can we avoid using _conda_ and instead install all requirements with [uv](https://docs.astral.sh/uv/)? Our base image comes with uv installed and a virtual environment set-up. However, it is suggested that you create your own virtual environment with uv, e.g., `uv venv -p 3.10 .venv310`
- The contents of the `meta.json` are used to populate the model card on our website under mhub.ai/models. The more information, the better.
- To test-build the model and to move forward with our [test routine](https://github.com/MHubAI/documentation/blob/main/documentation/mhub_contribution/testing_phase.md), an _mhub.toml_ file needs to be created.

locastre commented:
@LennyN95 Thank you for the in-depth feedback! We're working to address the suggestions.

I have an additional question: when we upload our test data to Zenodo (for the mhub.toml), should we refer to the mHub DOI# 13785615 or create a new DOI from an independent/new entry?

fedorov commented:
I would recommend renaming the PR to "SMIT CT Lung GTV segmentation model" (or something similar).

LennyN95 commented:
You're more than welcome. Thank you and your team for the great work.

> I have an additional question: when we upload our test data to Zenodo (for the mhub.toml), should we refer to the mHub DOI# 13785615 or create a new DOI from an independent/new entry?

@locastre We mainly use Zenodo for it's reproducible storage mechanism, so you can create a new DOI for your sample & reference.

LennyN95 commented:
@locastre FYI; There are also some errors in the [compliance check](https://github.com/MHubAI/models/actions/runs/13903139910/job/38899633433?pr=108) that need to be resolved:

> Errors:
> - Model folder models/msk_smit_lung_gtv does not contain a mhub.toml
>   https://github.com/MHubAI/documentation/blob/main/documentation/mhub_models/model_folder_structure.md
>   https://github.com/MHubAI/documentation/blob/main/documentation/mhub_contribution/testing_phase.md#test-procedure
> - Model meta json is not compliant with the schema: 'non-contrast' is a required property
>   https://github.com/MHubAI/documentation/blob/main/documentation/mhub_models/model_json.md
> - WORKDIR must not be set to any other than `/app` as defined in our base image. WORKDIR ${WORK_DIR}/msk_smit_lung_gtv/src
>   https://github.com/MHubAI/documentation/blob/main/documentation/mhub_models/the_mhub_dockerfile.md
>

locastre committed:
Update Dockerfile with venv python

locastre committed:
Create requirements.txt

locastre committed:
Update Dockerfile requirements path

locastre committed:
Update Dockerfile set BASH shell

locastre committed:
Update meta.json with draft submission comments

locastre committed:
Create mhub.toml

locastre committed:
Update meta.json

locastre committed:
Update meta.json

locastre committed:
Update mhub.toml Zenodo URL

locastre committed:
Update meta.json rename seg output to LUNG+NEOPLASM_MALIGNANT_PRIMARY

locastre committed:
Update SMITrunner.py name of output segmentation to LUNG+NEOPLASM_MALIGNANT_PRIMARY

locastre committed:
Rename SMITrunner.py to SMITRunner.py

locastre committed:
Update default.yml

locastre committed:
Update default.yml NiftiConverter parameters

locastre committed:
Create bash_run_SMIT_mhub.sh

locastre committed:
Update SMITRunner.py

locastre committed:
Update bash_run_SMIT_mhub.sh dos2unix line endings

locastre committed:
Update SMITRunner.py

locastre committed:
Delete models/msk_smit_lung_gtv/src/bash_run_SMIT_mhub.sh

locastre committed:
Delete models/msk_smit_lung_gtv/src/bash_run_SMIT_Segmentation.sh

locastre committed:
Update default.yml fix trailing typo in seg output name

locastre committed:
Update Dockerfile

locastre committed:
Create get_weights.sh

locastre committed:
Update Dockerfile run script to get weights

locastre committed:
Update mhub.toml

locastre committed:
Update meta.json

locastre committed:
Update meta.json

locastre committed:
Update Dockerfile

locastre committed:
Update meta.json correct tag to uri

locastre committed:
Update mhub.toml with Zenodo data from IDC

locastre committed:
Update meta.json remove overly long table

locastre commented:
@LennyN95 Hopefully we're ready for your review; we've addressed your feedback and tested the model within the mHub framework. Thanks for your help with this process.

LennyN95 commented:
![Static Badge](https://img.shields.io/badge/Failed-x?style=flat&logo=data%3Aimage%2Fsvg%2Bxml%3Bbase64%2CCjxzdmcgd2lkdGg9IjQxcHgiIGhlaWdodD0iNDJweCIgdmlld0JveD0iMCAwIDQxIDQyIiB2ZXJzaW9uPSIxLjEiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgeG1sbnM6eGxpbms9Imh0dHA6Ly93d3cudzMub3JnLzE5OTkveGxpbmsiPgogICAgPGcgaWQ9IndlYnNpdGUiIHN0cm9rZT0ibm9uZSIgc3Ryb2tlLXdpZHRoPSIxIiBmaWxsPSJub25lIiBmaWxsLXJ1bGU9ImV2ZW5vZGQiPgogICAgICAgIDxwYXRoIGQ9Ik01LjAzMiwwLjgyODU0MTc0MiBDNi4wNTg0MzQ2MSwwLjgyODU0MTc0MiA3LjA1NDUyMzM4LDEuMTc2NjcwNzggNy44NTc1MDAwNywxLjgxNjA0MjcxIEw3Ljg1NzUwMDA3LDEuODE2MDQyNzEgTDIwLjk5NiwxMi4yNzc1NDE3IEwzMy42NTQxOTc2LDIuMTk5MDI0MTUgQzM0LjMwMjAwNzIsMS42ODMyMDQzOCAzNS4wOTI2MjYyLDEuMzgwODM4MTEgMzUuOTE2NjQxOSwxLjMzMDEyNTQgTDM2LjE2NDczOTIsMS4zMjI0OTkzOSBDMzguMzE3MDE3NSwxLjMyMjQ5OTM5IDQwLjA3NDEzMjcsMy4wMDY0Njc4NiA0MC4xNzg2NDgyLDUuMTIyODU5MDUgTDQwLjE4MzU2NjUsNS4zMjI0OTkzOSBMNDAuMTgzNTY2NSwzNi42Nzc1MDA2IEM0MC4xODM1NjY1LDM3LjU4NTc4NjYgMzkuODcyOTg4NSwzOC40NjcwMjg2IDM5LjMwMjkxNjEsMzkuMTc2MjgwOCBDMzcuOTYyNjAwMSw0MC44NDM4MjY5IDM1LjU1NDY5NjYsNDEuMTYwNzYyNCAzMy44MzAwOTU0LDM5LjkzMzQ1NTIgTDMzLjY1NDE5NzYsMzkuODAwOTc1OCBMMjAuOTk3LDI5LjcyMDU0MTcgTDcuODU3NTAwMDcsNDAuMTgzOTU3MyBDNS45NTcxMDU1LDQxLjY5NzE1MDYgMy4yMTYxMTQ0LDQxLjQzODYyODEgMS42Mjg4NjQ2NywzOS42MzQ2MDYzIEwxLjQ4MzUwMDk3LDM5LjQ2MDk1ODMgQzAuODQ0MTI5MDQsMzguNjU3OTgxNiAwLjQ5NiwzNy42NjE4OTI5IDAuNDk2LDM2LjYzNTQ1ODMgTDAuNDk2LDM2LjYzNTQ1ODMgTDAuNDk2LDUuMzY0NTQxNzQgQzAuNDk2LDIuODU5Mzc4MTIgMi41MjY4MzYzNywwLjgyODU0MTc0MiA1LjAzMiwwLjgyODU0MTc0MiBaIE01LjAzMiwxLjgzNjU0MTc0IEMzLjA4MzUzOTQsMS44MzY1NDE3NCAxLjUwNCwzLjQxNjA4MTE0IDEuNTA0LDUuMzY0NTQxNzQgTDEuNTA0LDUuMzY0NTQxNzQgTDEuNTA0LDM2LjYzNTQ1ODMgQzEuNTA0LDM3LjQzMzc5NjMgMS43NzQ3NjcwMywzOC4yMDg1MzIgMi4yNzIwNTYzMSwzOC44MzMwNjk0IEMzLjQ4NTc2MzQ0LDQwLjM1NzM0NDIgNS43MDUzMzYzNSw0MC42MDkxMDkxIDcuMjI5NjExMTYsMzkuMzk1NDAxOSBMNy4yMjk2MTExNiwzOS4zOTU0MDE5IEwyNi44NjU5MjU5LDIzLjc1OTk0MzcgQzI3LjA3Mzk2NDcsMjMuNTk0MjkyMyAyNy4yNjI2MDcsMjMuNDA1NjUgMjcuNDI4MjU4NCwyMy4xOTc2MTEyIEMyOC42NDE5NjU1LDIxLjY3MzMzNjQgMjguMzkwMjAwNywxOS40NTM3NjM0IDI2Ljg2NTkyNTksMTguMjQwMDU2MyBMMjYuODY1OTI1OSwxOC4yNDAwNTYzIEw3LjIyOTYxMTE2LDIuNjA0NTk4MDUgQzYuNjA1MDczNzQsMi4xMDczMDg3NyA1LjgzMDMzODAzLDEuODM2NTQxNzQgNS4wMzIsMS44MzY1NDE3NCBaIE01LjU0OTY0NDc2LDUuMTQyNjMxMDMgTDUuNjQxNjIxMyw1LjIwNzQxNDU2IEwyNC43MzEwNTExLDIwLjIwNzQxNDYgQzI1LjE2ODc4NCwyMC41NTEzNzQyIDI1LjI0NDgwMjIsMjEuMTg1MDYxMSAyNC45MDA4NDI2LDIxLjYyMjc5NCBDMjQuNTgzMzQxNCwyMi4wMjY4NTUxIDI0LjAxODk3MiwyMi4xMjI3MDk2IDIzLjU4OTczNTksMjEuODY0NTMyIEwyMy40ODU0NjMyLDIxLjc5MjU4NTQgTDYuMDI3LDguMDc0IEw2LjAyNjgyNzMzLDM1IEM2LjAyNjgyNzMzLDM1LjUxOTU4OTUgNS42MzM2OTc1MywzNS45NDczNTg2IDUuMTI4NjYwMDQsMzYuMDAyMDg1MiBMNS4wMTg4MjczMywzNi4wMDggQzQuNDk5MjM3ODQsMzYuMDA4IDQuMDcxNDY4NzQsMzUuNjE0ODcwMiA0LjAxNjc0MjE0LDM1LjEwOTgzMjcgTDQuMDEwODI3MzMsMzUgTDQuMDEwODI3MzMsNiBDNC4wMTA4MjczMyw1LjE5NzEwODMzIDQuODk0MTIyODQsNC43MzEzMjEwNCA1LjU0OTY0NDc2LDUuMTQyNjMxMDMgWiIgaWQ9IkNvbWJpbmVkLVNoYXBlIiBmaWxsPSIjRkZGRkZGIj48L3BhdGg%2BCiAgICA8L2c%2BCjwvc3ZnPg%3D%3D&logoColor=blue&label=Model%20Test&labelColor=%23313F49&color=red)

<details><summary>Test Results</summary>

```yaml
id: 7e53fc04-004d-4f0e-83cb-803fc3bd4b8f
name: MHub Test Report (default)
date: '2025-04-22 12:06:20'
missing_files:
- seg.dcm
summary:
  files_missing: 1
  files_extra: 0
  checks: {}
conclusion: false

```

</details>

LennyN95 commented:
Thank you Jue for the great work!

Unfortunately, the test failed due to what looks like a GPU incompatibility (see below). We use NVIDIA RTX A6000 GPU for testing and so far all models worked fine - can you have a look? Ideally we can find a solution that applies to a wide range of customer hardware.

```text
/app/.venv39/lib/python3.9/site-packages/torch/functional.py:478: UserWarning: torch.meshgrid: in an upcoming release, it will be required to pass the indexing argument. (Triggered internally at  ../aten/src/ATen/native/TensorShape.cpp:2895.)
return _VF.meshgrid(tensors, **kwargs)  # type: ignore[attr-defined]
/app/.venv39/lib/python3.9/site-packages/torch/cuda/__init__.py:146: UserWarning:
NVIDIA RTX A6000 with CUDA capability sm_86 is not compatible with the current PyTorch installation.
The current PyTorch install supports CUDA capabilities sm_37 sm_50 sm_60 sm_70.
If you want to use the NVIDIA RTX A6000 GPU with PyTorch, please check the instructions at https://pytorch.org/get-started/locally/

warnings.warn(incompatible_device_warn.format(device_name, capability, " ".join(arch_list), device_name))
val data size is  1
info: started to load weight:  trained_weights/model_Mhub.pt
info: model emb feature is :  96
info: Successfully loaded trained weights:  trained_weights/model_Mhub.pt
Traceback (most recent call last):
File "/app/models/msk_smit_lung_gtv/src/run_segmentation.py", line 252, in <module>
main()
File "/app/models/msk_smit_lung_gtv/src/run_segmentation.py", line 218, in main
val_data["pred"] = sliding_window_inference(val_inputs,
File "/app/models/msk_smit_lung_gtv/src/edit_inference_utils.py", line 120, in sliding_window_inference
importance_map = compute_importance_map(
File "/app/.venv39/lib/python3.9/site-packages/monai/data/utils.py", line 763, in compute_importance_map
importance_map = torch.ones(patch_size, device=device).float()
RuntimeError: CUDA error: no kernel image is available for execution on the device
CUDA kernel errors might be asynchronously reported at some other API call,so the stacktrace below might be incorrect.
For debugging consider passing CUDA_LAUNCH_BLOCKING=1.
Done in 19.371 seconds.
```

locastre committed:
Update Dockerfile explicit install instruction for torch 1.12.1+cu116

locastre commented:
@LennyN95 Hi Lenny, I added an explicit install command for `torch==1.12.1+cu116` which should properly support `sm_86`. Let us know if this works. Thanks!

LennyN95 commented:
![Static Badge](https://img.shields.io/badge/Failed-x?style=flat&logo=data%3Aimage%2Fsvg%2Bxml%3Bbase64%2CCjxzdmcgd2lkdGg9IjQxcHgiIGhlaWdodD0iNDJweCIgdmlld0JveD0iMCAwIDQxIDQyIiB2ZXJzaW9uPSIxLjEiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgeG1sbnM6eGxpbms9Imh0dHA6Ly93d3cudzMub3JnLzE5OTkveGxpbmsiPgogICAgPGcgaWQ9IndlYnNpdGUiIHN0cm9rZT0ibm9uZSIgc3Ryb2tlLXdpZHRoPSIxIiBmaWxsPSJub25lIiBmaWxsLXJ1bGU9ImV2ZW5vZGQiPgogICAgICAgIDxwYXRoIGQ9Ik01LjAzMiwwLjgyODU0MTc0MiBDNi4wNTg0MzQ2MSwwLjgyODU0MTc0MiA3LjA1NDUyMzM4LDEuMTc2NjcwNzggNy44NTc1MDAwNywxLjgxNjA0MjcxIEw3Ljg1NzUwMDA3LDEuODE2MDQyNzEgTDIwLjk5NiwxMi4yNzc1NDE3IEwzMy42NTQxOTc2LDIuMTk5MDI0MTUgQzM0LjMwMjAwNzIsMS42ODMyMDQzOCAzNS4wOTI2MjYyLDEuMzgwODM4MTEgMzUuOTE2NjQxOSwxLjMzMDEyNTQgTDM2LjE2NDczOTIsMS4zMjI0OTkzOSBDMzguMzE3MDE3NSwxLjMyMjQ5OTM5IDQwLjA3NDEzMjcsMy4wMDY0Njc4NiA0MC4xNzg2NDgyLDUuMTIyODU5MDUgTDQwLjE4MzU2NjUsNS4zMjI0OTkzOSBMNDAuMTgzNTY2NSwzNi42Nzc1MDA2IEM0MC4xODM1NjY1LDM3LjU4NTc4NjYgMzkuODcyOTg4NSwzOC40NjcwMjg2IDM5LjMwMjkxNjEsMzkuMTc2MjgwOCBDMzcuOTYyNjAwMSw0MC44NDM4MjY5IDM1LjU1NDY5NjYsNDEuMTYwNzYyNCAzMy44MzAwOTU0LDM5LjkzMzQ1NTIgTDMzLjY1NDE5NzYsMzkuODAwOTc1OCBMMjAuOTk3LDI5LjcyMDU0MTcgTDcuODU3NTAwMDcsNDAuMTgzOTU3MyBDNS45NTcxMDU1LDQxLjY5NzE1MDYgMy4yMTYxMTQ0LDQxLjQzODYyODEgMS42Mjg4NjQ2NywzOS42MzQ2MDYzIEwxLjQ4MzUwMDk3LDM5LjQ2MDk1ODMgQzAuODQ0MTI5MDQsMzguNjU3OTgxNiAwLjQ5NiwzNy42NjE4OTI5IDAuNDk2LDM2LjYzNTQ1ODMgTDAuNDk2LDM2LjYzNTQ1ODMgTDAuNDk2LDUuMzY0NTQxNzQgQzAuNDk2LDIuODU5Mzc4MTIgMi41MjY4MzYzNywwLjgyODU0MTc0MiA1LjAzMiwwLjgyODU0MTc0MiBaIE01LjAzMiwxLjgzNjU0MTc0IEMzLjA4MzUzOTQsMS44MzY1NDE3NCAxLjUwNCwzLjQxNjA4MTE0IDEuNTA0LDUuMzY0NTQxNzQgTDEuNTA0LDUuMzY0NTQxNzQgTDEuNTA0LDM2LjYzNTQ1ODMgQzEuNTA0LDM3LjQzMzc5NjMgMS43NzQ3NjcwMywzOC4yMDg1MzIgMi4yNzIwNTYzMSwzOC44MzMwNjk0IEMzLjQ4NTc2MzQ0LDQwLjM1NzM0NDIgNS43MDUzMzYzNSw0MC42MDkxMDkxIDcuMjI5NjExMTYsMzkuMzk1NDAxOSBMNy4yMjk2MTExNiwzOS4zOTU0MDE5IEwyNi44NjU5MjU5LDIzLjc1OTk0MzcgQzI3LjA3Mzk2NDcsMjMuNTk0MjkyMyAyNy4yNjI2MDcsMjMuNDA1NjUgMjcuNDI4MjU4NCwyMy4xOTc2MTEyIEMyOC42NDE5NjU1LDIxLjY3MzMzNjQgMjguMzkwMjAwNywxOS40NTM3NjM0IDI2Ljg2NTkyNTksMTguMjQwMDU2MyBMMjYuODY1OTI1OSwxOC4yNDAwNTYzIEw3LjIyOTYxMTE2LDIuNjA0NTk4MDUgQzYuNjA1MDczNzQsMi4xMDczMDg3NyA1LjgzMDMzODAzLDEuODM2NTQxNzQgNS4wMzIsMS44MzY1NDE3NCBaIE01LjU0OTY0NDc2LDUuMTQyNjMxMDMgTDUuNjQxNjIxMyw1LjIwNzQxNDU2IEwyNC43MzEwNTExLDIwLjIwNzQxNDYgQzI1LjE2ODc4NCwyMC41NTEzNzQyIDI1LjI0NDgwMjIsMjEuMTg1MDYxMSAyNC45MDA4NDI2LDIxLjYyMjc5NCBDMjQuNTgzMzQxNCwyMi4wMjY4NTUxIDI0LjAxODk3MiwyMi4xMjI3MDk2IDIzLjU4OTczNTksMjEuODY0NTMyIEwyMy40ODU0NjMyLDIxLjc5MjU4NTQgTDYuMDI3LDguMDc0IEw2LjAyNjgyNzMzLDM1IEM2LjAyNjgyNzMzLDM1LjUxOTU4OTUgNS42MzM2OTc1MywzNS45NDczNTg2IDUuMTI4NjYwMDQsMzYuMDAyMDg1MiBMNS4wMTg4MjczMywzNi4wMDggQzQuNDk5MjM3ODQsMzYuMDA4IDQuMDcxNDY4NzQsMzUuNjE0ODcwMiA0LjAxNjc0MjE0LDM1LjEwOTgzMjcgTDQuMDEwODI3MzMsMzUgTDQuMDEwODI3MzMsNiBDNC4wMTA4MjczMyw1LjE5NzEwODMzIDQuODk0MTIyODQsNC43MzEzMjEwNCA1LjU0OTY0NDc2LDUuMTQyNjMxMDMgWiIgaWQ9IkNvbWJpbmVkLVNoYXBlIiBmaWxsPSIjRkZGRkZGIj48L3BhdGg%2BCiAgICA8L2c%2BCjwvc3ZnPg%3D%3D&logoColor=blue&label=Model%20Test&labelColor=%23313F49&color=red)

<details><summary>Test Results</summary>

```yaml
id: aa683d35-ae95-4dd2-bdb9-4c7bc81cfe9c
name: MHub Test Report (default)
date: '2025-04-23 09:37:41'
missing_files:
- seg.dcm
extra_files:
- 1.3.6.1.4.1.14519.5.2.1.1.11635178980898764572976586249071182079/smit.seg.dcm
summary:
  files_missing: 1
  files_extra: 1
  checks: {}
conclusion: false

```

</details>

LennyN95 commented:
Thank you @locastre. The model now runs, we're almost there! The test is still failing, because the reference data doesn't match with the model output (see report). Please make sure, that the reference data looks like:

```text
data/output_data/
`-- 1.3.6.1.4.1.14519.5.2.1.1.11635178980898764572976586249071182079
    `-- smit.seg.dcm
```

You can follow [the steps here ](https://github.com/MHubAI/documentation/blob/main/documentation/mhub_contribution/testing_phase.md#test-procedure)to prepare the output for the test procedure.

To increase the usability, could you change the filename from `smit.seg.dcm` to `<model_name>.seg.dcm`? We do not enforce this convention but it would make downstream automation a lot more convenient. Alternatively, rename the model and the folder consistently to `smit`.

locastre committed:
Update get_weights.sh

locastre committed:
Update default.yml

locastre committed:
Update mhub.toml doi

locastre commented:
@LennyN95 thanks for the catch. I've updated the data in the test zip and renamed the output SEG to `msk_smit_lung_gtv.seg.dcm` in `default.yml`, as you suggested.

locastre committed:
Update mhub.toml link to Zenodo test data, 3x the charm

LennyN95 commented:
![Static Badge](https://img.shields.io/badge/Passed-x?style=flat&logo=data%3Aimage%2Fsvg%2Bxml%3Bbase64%2CCjxzdmcgd2lkdGg9IjQxcHgiIGhlaWdodD0iNDJweCIgdmlld0JveD0iMCAwIDQxIDQyIiB2ZXJzaW9uPSIxLjEiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgeG1sbnM6eGxpbms9Imh0dHA6Ly93d3cudzMub3JnLzE5OTkveGxpbmsiPgogICAgPGcgaWQ9IndlYnNpdGUiIHN0cm9rZT0ibm9uZSIgc3Ryb2tlLXdpZHRoPSIxIiBmaWxsPSJub25lIiBmaWxsLXJ1bGU9ImV2ZW5vZGQiPgogICAgICAgIDxwYXRoIGQ9Ik01LjAzMiwwLjgyODU0MTc0MiBDNi4wNTg0MzQ2MSwwLjgyODU0MTc0MiA3LjA1NDUyMzM4LDEuMTc2NjcwNzggNy44NTc1MDAwNywxLjgxNjA0MjcxIEw3Ljg1NzUwMDA3LDEuODE2MDQyNzEgTDIwLjk5NiwxMi4yNzc1NDE3IEwzMy42NTQxOTc2LDIuMTk5MDI0MTUgQzM0LjMwMjAwNzIsMS42ODMyMDQzOCAzNS4wOTI2MjYyLDEuMzgwODM4MTEgMzUuOTE2NjQxOSwxLjMzMDEyNTQgTDM2LjE2NDczOTIsMS4zMjI0OTkzOSBDMzguMzE3MDE3NSwxLjMyMjQ5OTM5IDQwLjA3NDEzMjcsMy4wMDY0Njc4NiA0MC4xNzg2NDgyLDUuMTIyODU5MDUgTDQwLjE4MzU2NjUsNS4zMjI0OTkzOSBMNDAuMTgzNTY2NSwzNi42Nzc1MDA2IEM0MC4xODM1NjY1LDM3LjU4NTc4NjYgMzkuODcyOTg4NSwzOC40NjcwMjg2IDM5LjMwMjkxNjEsMzkuMTc2MjgwOCBDMzcuOTYyNjAwMSw0MC44NDM4MjY5IDM1LjU1NDY5NjYsNDEuMTYwNzYyNCAzMy44MzAwOTU0LDM5LjkzMzQ1NTIgTDMzLjY1NDE5NzYsMzkuODAwOTc1OCBMMjAuOTk3LDI5LjcyMDU0MTcgTDcuODU3NTAwMDcsNDAuMTgzOTU3MyBDNS45NTcxMDU1LDQxLjY5NzE1MDYgMy4yMTYxMTQ0LDQxLjQzODYyODEgMS42Mjg4NjQ2NywzOS42MzQ2MDYzIEwxLjQ4MzUwMDk3LDM5LjQ2MDk1ODMgQzAuODQ0MTI5MDQsMzguNjU3OTgxNiAwLjQ5NiwzNy42NjE4OTI5IDAuNDk2LDM2LjYzNTQ1ODMgTDAuNDk2LDM2LjYzNTQ1ODMgTDAuNDk2LDUuMzY0NTQxNzQgQzAuNDk2LDIuODU5Mzc4MTIgMi41MjY4MzYzNywwLjgyODU0MTc0MiA1LjAzMiwwLjgyODU0MTc0MiBaIE01LjAzMiwxLjgzNjU0MTc0IEMzLjA4MzUzOTQsMS44MzY1NDE3NCAxLjUwNCwzLjQxNjA4MTE0IDEuNTA0LDUuMzY0NTQxNzQgTDEuNTA0LDUuMzY0NTQxNzQgTDEuNTA0LDM2LjYzNTQ1ODMgQzEuNTA0LDM3LjQzMzc5NjMgMS43NzQ3NjcwMywzOC4yMDg1MzIgMi4yNzIwNTYzMSwzOC44MzMwNjk0IEMzLjQ4NTc2MzQ0LDQwLjM1NzM0NDIgNS43MDUzMzYzNSw0MC42MDkxMDkxIDcuMjI5NjExMTYsMzkuMzk1NDAxOSBMNy4yMjk2MTExNiwzOS4zOTU0MDE5IEwyNi44NjU5MjU5LDIzLjc1OTk0MzcgQzI3LjA3Mzk2NDcsMjMuNTk0MjkyMyAyNy4yNjI2MDcsMjMuNDA1NjUgMjcuNDI4MjU4NCwyMy4xOTc2MTEyIEMyOC42NDE5NjU1LDIxLjY3MzMzNjQgMjguMzkwMjAwNywxOS40NTM3NjM0IDI2Ljg2NTkyNTksMTguMjQwMDU2MyBMMjYuODY1OTI1OSwxOC4yNDAwNTYzIEw3LjIyOTYxMTE2LDIuNjA0NTk4MDUgQzYuNjA1MDczNzQsMi4xMDczMDg3NyA1LjgzMDMzODAzLDEuODM2NTQxNzQgNS4wMzIsMS44MzY1NDE3NCBaIE01LjU0OTY0NDc2LDUuMTQyNjMxMDMgTDUuNjQxNjIxMyw1LjIwNzQxNDU2IEwyNC43MzEwNTExLDIwLjIwNzQxNDYgQzI1LjE2ODc4NCwyMC41NTEzNzQyIDI1LjI0NDgwMjIsMjEuMTg1MDYxMSAyNC45MDA4NDI2LDIxLjYyMjc5NCBDMjQuNTgzMzQxNCwyMi4wMjY4NTUxIDI0LjAxODk3MiwyMi4xMjI3MDk2IDIzLjU4OTczNTksMjEuODY0NTMyIEwyMy40ODU0NjMyLDIxLjc5MjU4NTQgTDYuMDI3LDguMDc0IEw2LjAyNjgyNzMzLDM1IEM2LjAyNjgyNzMzLDM1LjUxOTU4OTUgNS42MzM2OTc1MywzNS45NDczNTg2IDUuMTI4NjYwMDQsMzYuMDAyMDg1MiBMNS4wMTg4MjczMywzNi4wMDggQzQuNDk5MjM3ODQsMzYuMDA4IDQuMDcxNDY4NzQsMzUuNjE0ODcwMiA0LjAxNjc0MjE0LDM1LjEwOTgzMjcgTDQuMDEwODI3MzMsMzUgTDQuMDEwODI3MzMsNiBDNC4wMTA4MjczMyw1LjE5NzEwODMzIDQuODk0MTIyODQsNC43MzEzMjEwNCA1LjU0OTY0NDc2LDUuMTQyNjMxMDMgWiIgaWQ9IkNvbWJpbmVkLVNoYXBlIiBmaWxsPSIjRkZGRkZGIj48L3BhdGg%2BCiAgICA8L2c%2BCjwvc3ZnPg%3D%3D&logoColor=blue&label=Model%20Test&labelColor=%23313F49)

<details><summary>Test Results</summary>

```yaml
id: d5b328ee-037e-4909-9827-65e6e1c31a67
name: MHub Test Report (default)
date: '2025-04-23 19:08:01'
checked_files:
- file: msk_smit_lung_gtv.seg.dcm
  path: /app/data/output_data/1.3.6.1.4.1.14519.5.2.1.1.11635178980898764572976586249071182079/msk_smit_lung_gtv.seg.dcm
  checks:
  - checker: DicomsegContentCheck
    notes:
    - label: Segment Count
      description: The number of segments identified in the inspected dicomseg file.
      info: 1
summary:
  files_missing: 0
  files_extra: 0
  checks:
    DicomsegContentCheck:
      files: 1
conclusion: true

```

</details>

LennyN95 commented:
@locastre Amazing, we now passed all tests - yay! Thank you for the great work!
Do you have 1-5 example images you would like to have on the website? You can post them here :)

locastre commented:
![SceneView](https://github.com/user-attachments/assets/2859add8-25b5-406c-808c-4d8503b026e2)
@LennyN95 Jue provided this screencap of representative segmentation. Will this do?

LennyN95 commented:
@locastre, perfect! I have one last request that I initially overlooked. Could you add the [sample annotation](https://github.com/MHubAI/documentation/blob/main/documentation/mhubio/the_mhubio_config_file.md#sample-data-annotation) to the default.yaml workflow? This addition will help our users understand how the input and output data are organized. It also gives us a chance to explain the role of each file and folder, which is quite self-explanatory in the case of DICOM. This is a fairly new feature I added, and we are gradually rolling it out for all legacy models as well. You can [look at this model for more examples](https://github.com/MHubAI/models/tree/main/models/fmcib_radiomics/config).

locastre committed:
Update default.yml with sample input/output file structure

locastre committed:
Update default.yml

locastre commented:
@LennyN95 `sample` annotation added to end of `default.yml`

LennyN95 commented:
@locastre The model is now online. Amazing work!! FYI; I shortened the title to `SMIT Self-supervised Lung GTV Segmentation`, please let me know if you prefer an alternative version.
